apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -219,5 +282,185 @@ public WriteTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHelpe
+                        new SpannerParameter("Float64Value", SpannerDbType.Float64, double.NegativeInfinity)
+                    }));
+            using (var reader = await GetLastRowAsync())
+            {
+                Assert.True(double.IsNegativeInfinity(reader.GetFieldValue<double>("Float64Value")));
+            }
+        }
+
+        [Fact]
+        public async Task WriteRandomBytes()
+        {
+            var seedByte = (byte)(Environment.TickCount % 256);
+            var rnd = new Random(seedByte);
+
+            //we write 1-50 rows where each row contains a bytearray of size 1-50
+            //whose bytes are randomly generated with the given seed.
+            //The seed itself is written as the first row in an array of size=1.
+            var recordedValues = new Dictionary<string, byte[]>();
+
+            await InsertRecordedBytesAsync(new[] {seedByte}, recordedValues);
+
+            var numRows = rnd.Next(50);
+            for (var i = 0; i < numRows; i++)
+            {
+                var byteArray = new byte[rnd.Next(50)];
+                rnd.NextBytes(byteArray);
+                await InsertRecordedBytesAsync(byteArray, recordedValues);
+            }
+
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                string sqlQuery = $@"SELECT K,BytesValue
+                                FROM T
+                                WHERE K IN ({string.Join(", ", recordedValues.Keys.Select(x => $"'{x}'"))})";
+
+                var cmd = connection.CreateSelectCommand(sqlQuery);
+                using (var reader = (SpannerDataReader) await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        var key = reader.GetFieldValue<string>("K");
+                        var value = reader.GetFieldValue<byte[]>("BytesValue");
+                        Assert.Equal(recordedValues[key], value);
+                        recordedValues.Remove(key);
+                    }
+                }
+                Assert.Equal(0, recordedValues.Count);
+            }
+        }
+
+        [Fact]
+        public async Task BadTableName()
+        {
+            int rowsWritten = 0;
+            var exceptionCaught = false;
+
+            try
+            {
+                using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+                {
+                    var cmd = connection.CreateInsertCommand("badjuju", new SpannerParameterCollection
+                    {
+                        new SpannerParameter("K", SpannerDbType.String, UniqueString())
+                    });
+
+                    rowsWritten = await cmd.ExecuteNonQueryAsync();
+                }
+            }
+            catch (SpannerException e)
+            {
+                exceptionCaught = true;
+                Logger.Instance.Debug($"BadTableName: Caught error code:{e.ErrorCode}");
```

Comment:
i found it helpful when debugging issues.  the log shows up next to the test when running inside VS with logging on.